spec/runner_spec.rb

```diff
@@ -16,11 +16,37 @@ def on_message(&block)
   def process(message)
     @messages << message
 
-    processor = @processor_queue.shift || proc {} 
+    processor = @processor_queue.shift || proc {}
     processor.call(message)
   end
 end
 
+class TestBatchConsumer < Racecar::Consumer
+  attr_reader :messages
+
+  def initialize
+    @messages = []
+    @processor_queue = []
+  end
+
+  def on_message(&block)
+    @processor_queue << block
+    self
+  end
+
+  def process_batch(messages)
+    @messages += messages
```

Comment:
Actually, ruby-kafka doesn't pass an array of messages to the block in `#each_batch` – it passes a [FetchedBatch](http://www.rubydoc.info/gems/ruby-kafka/Kafka/FetchedBatch) object that contains a `#messages` method as well as other metadata that's actually useful. I think `#process_batch` should be called with that batch object.